### ISSUE TITLE
Rule proposal: Always wrap `JSON.parse` in `try`/`catch` block

### ISSUE THREAD

**Issue by niksy** [sindresorhus/eslint-plugin-unicorn#1543]
When dealing with external and unknown data and parsing with `JSON.parse`, there can be situations where parsing can fail and can lead to uncaught errors in codebase. This rule would flag every `JSON.parse` which doesn’t have (at least one?) `try`/`catch` block as ancestor. Rule *could* potentially be fixable.

## Fail

```js
const response = undefined;
const data = JSON.parse(response);
```

## Pass

```js
const response = undefined;
let data;
try {
  data = JSON.parse(response);
} catch {
  data = {};
}
```
